### PR TITLE
Update dns_dhcp.rst

### PR DESCRIPTION
**Pull request by cotosso**
added dns rebind section

### CODE REVIEW

File: dns_dhcp.rst [hunk 1]
```diff
@@ -112,6 +112,53 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
 
 This is true also for wildcards: if both specific and wildcard domains are defined for the same pattern, the specific one takes precedence (e.g., having ``/google.com/`` and ``/*google.com/`` : the first will handle google.com and www.google.com, the wildcard will handle supergoogle.com.
 
+DNS Rebind Protection
+^^^^^^^^^^^^^^^^^^
```
Comment (gsanchietti) [inline dns_dhcp.rst:116]:
```suggestion
DNS Rebind Protection
^^^^^^^^^^^^^^^^^^^^^^
```

File: dns_dhcp.rst [hunk 2]
```diff
@@ -112,6 +112,53 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+
+DNS Rebind Protection is a security feature that safeguards against DNS rebinding attacks. It blocks the use of private IP ranges by public domains, preventing malicious websites from manipulating browsers to make unauthorized requests to local network devices.
+
+DNS Rebind Protection is enabled by default on NethSecurity and usually does not have operational repercussions. 
+In the presence of split DNS, resolving public domains with internal resources, rebind protection may lead to resolution issues. In such scenarios, potential problems can be checked in the logs, where lines similar to these may appear:
```
Comment (gsanchietti) [inline dns_dhcp.rst:121]:
```suggestion
In the presence of split DNS, resolving public domains with internal resources, rebind protection may lead to resolution issues. In such scenarios, potential problems can be found in the log (`/var/log/messages`), where lines similar to these may appear:
```

**Review by gsanchietti — COMMENTED** [review]
Do not merge when ready: I need to build it because there are some issues with title nesting level.

File: dns_dhcp.rst [hunk 3]
```diff
@@ -112,6 +112,53 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can easily fix any of these issues from the CLI.
+
+Solution 1: Put the specific domain in a whitelist (suggested)::
+
+  uci set dhcp.@dnsmasq[0].rebind_domain='nethesis.it'
+
+in case if multiple domains write them separated by a space::
+
+ uci set dhcp.@dnsmasq[0].rebind_domain='nethesis.it' 'mydomain'
```
Comment (filippocarletti) [inline dns_dhcp.rst:137]: AFAIK this syntax is invalid.
It's be:
uci add_list dhcp.@dnsmasq[0].rebind_domain="nethesis.it"
uci add_list dhcp.@dnsmasq[0].rebind_domain="mydomain"

File: dns_dhcp.rst [hunk 4]
```diff
@@ -112,6 +112,53 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+
+ uci set dhcp.@dnsmasq[0].rebind_protection='0'
+ uci commit dhcp
+ /etc/init.d/dnsmasq restart
+
+How to enable DNS rebind protection (e.g. on migrated firewalls)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+
+ uci set dhcp.@dnsmasq[0].rebind_protection='1'
```
Comment (filippocarletti) [inline dns_dhcp.rst:156]: What about rebind_localhost=0 to be safe?

File: dns_dhcp.rst [hunk 5]
```diff
@@ -112,6 +112,53 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+
+.. note:: To ensure maximum compatibility and prevent malfunctions in migrated installations using the dedicated tool from NethServer 7.9, DNS Rebind Protection is disabled, ensuring the same behavior as the previous version.
+
+How to fix DNS rebind protection issues
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can easily fix any of these issues from the CLI.
+
+Solution 1: Put the specific domain in a whitelist (suggested)::
+
+  uci set dhcp.@dnsmasq[0].rebind_domain='nethesis.it'
```
Comment (cotosso) [inline dns_dhcp.rst:133]:
```suggestion
 uci add_list dhcp.@dnsmasq[0].rebind_domain="nethesis.it"
```

File: dns_dhcp.rst (hunk 4 above)
Comment (filippocarletti) [reply dns_dhcp.rst:152]: No, I mean to suggest to set rebind_localhost=1 when enabling rebind protection.

> Allows upstream 127.0.0.0/8 responses, required for DNS based blacklist services, only takes effect if rebind protection is enabled

Comment (cotosso) [reply dns_dhcp.rst:153]: I added rebind_localhost information @filippocarletti

File: dns_dhcp.rst [hunk 6]
```diff
@@ -112,6 +112,51 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+Completely disable DNS rebind protection using these commands::
+
+ uci set dhcp.@dnsmasq[0].rebind_protection='0'
+ uci commit dhcp
+ /etc/init.d/dnsmasq restart
+
+How to enable DNS rebind protection (e.g. on migrated firewalls)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+If you have disabled rebind protection in the past or if your configuration comes from a migration and you want to enable rebind protection, we suggest enabling `rebind_localhost` as well. This parameter only takes effect if rebind protection is enabled and allows upstream 127.0.0.0/8 responses, which are required for DNS-based blacklist services.
```
Comment (cotosso) [inline dns_dhcp.rst:151]:
```suggestion
If you have previously disabled rebind protection or if your configuration has been migrated and you wish to enable rebind protection, it is recommended to also activate the `rebind_localhost` parameter. This setting takes effect exclusively when rebind protection is enabled and permits upstream responses from 127.0.0.0/8, essential for DNS-based blacklist services.
```

File: dns_dhcp.rst [hunk 7]
```diff
@@ -112,6 +112,51 @@ NethSecurity will send queries for google.com and gmail.google.com to 1.2.3.4, b
...
+Completely disable DNS rebind protection using these commands::
+
+ uci set dhcp.@dnsmasq[0].rebind_protection='0'
+ uci commit dhcp
+ /etc/init.d/dnsmasq restart
+
+How to enable DNS rebind protection (e.g. on migrated firewalls)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+::
+If you have previously disabled rebind protection or if your configuration has been migrated and you wish to enable rebind protection, it is recommended to also activate the `rebind_localhost` parameter. This setting takes effect exclusively when rebind protection is enabled and permits upstream responses from 127.0.0.0/8, essential for DNS-based blacklist services.
```
Comment (cotosso) [inline dns_dhcp.rst:151]:
```suggestion
If you have previously disabled rebind protection or if your configuration comes from a migration and you wish to enable rebind protection, it is recommended to also activate the `rebind_localhost` parameter. This setting takes effect exclusively when rebind protection is enabled and permits upstream responses from 127.0.0.0/8, essential for DNS-based blacklist services.
```

**Comment by gsanchietti** [comment]
The format style is quite messed up.
I will fix it and manually merge the PR

**Comment by gsanchietti** [comment]
Manually merged: https://github.com/NethServer/nethsecurity-docs/commit/ed7c484521f633e8cae9810829eb9b9bddb658d6

**Pull request closed without merging**